### DISCUSSION TITLE
Possible to read state of capslock key using KeyboardBT?

### DISCUSSION THREAD

**Discussion by itrr0** [earlephilhower/arduino-pico#2331] [Q&A] [answered]
I'm porting a project from the mbed os core for rp2040 to the arduino-pico core since it has a bluetooth keyboard library and I'm wondering if it is possible to see if the capslock key is activated or disabled on the connected bluetooth device.

**Comment by earlephilhower** [comment, marked as answer]
Ooops, mybad.  I completely zoned out and forgot we already support it.  The `onLED` callback works in USB and in Bluetooth mode.  See the KeyboardBT example. You can't "read" the state, but you get a callback whenever the state changes, so just keep track of it there.  In the example it sets the Pico's LED on or off for CAPSLOCK.

**Reply by itrr0** [reply]
Thank you so much, I thought that function only kept track of capslock presses sent from the microcontroller but it appears that I was wrong.👍
